src/test/java/nurseybook/testutil/TypicalTasks.java

```diff
@@ -85,17 +98,33 @@
      */
     public static NurseyBook getTypicalNurseyBook() {
         NurseyBook nb = new NurseyBook();
-        for (Task t : getTypicalTasks()) {
-            nb.addTask(t);
-        }
-        for (Elderly elderly : TypicalElderlies.getTypicalElderlies()) {
-            nb.addElderly(elderly);
-        }
+        nb.setElderlies(getTypicalElderlyBuilders().stream().map(s -> s.build()).collect(Collectors.toList()));
+        nb.setTasks(getTypicalTaskBuilders().stream().map(s -> s.build()).collect(Collectors.toList()));
+        nb.updateRecurringTasksDate();
+        nb.updateTasksOverdueStatus();
+        nb.reorderTasksChronologically();
         return nb;
     }
```

Comment:
Instead of returning NurseyBook objects that all link to the same elderly/task objects, this returns NurseyBook objects that create new copies of those default Elderly and Task objects.

Reply:
This is to prevent state modifications from leaking from one test to another